integration/singularity/store.go

```diff
@@ -431,3 +436,114 @@ func (s *SingularityStore) Describe(ctx context.Context, id blob.ID) (*blob.Desc
 	}
 	return descriptor, nil
 }
+
+func (s *SingularityStore) runCleanupWorker(ctx context.Context) {
+	// Run immediately once before starting ticker
+	go func() {
+		if err := s.cleanup(context.Background()); err != nil {
+			logger.Errorf("Local store cleanup failed: %w", err)
+		}
+	}()
+
+	ticker := time.NewTicker(s.localCleanupInterval)
+cleanupLoop:
+	for {
+		select {
+		case <-ticker.C:
+			go func() {
+				if err := s.cleanup(context.Background()); err != nil {
+					logger.Errorf("Local store cleanup failed: %w", err)
+				}
+			}()
+		case <-s.closing:
+			break cleanupLoop
+		}
+	}
+}
+
+var errCleanupAlreadyRunning = errors.New("cleanup already running")
+
+func (s *SingularityStore) cleanup(ctx context.Context) error {
+	if s.cleanupActive.Load() {
+		return errCleanupAlreadyRunning
+	}
+
+	// Mark cleanup active for the duration of the function
+	s.cleanupActive.Store(true)
+	defer s.cleanupActive.Store(false)
+
+	logger.Infof("Starting local store cleanup...")
+
+	dir, err := os.ReadDir(s.local.Dir())
```

Comment:
I am curious if using `filepath.Walk` would make the search logic simpler?

Reply:
it's only scanning one layer of a directory, so i don't think so? unless i misunderstood how the local store is laid out